clients/da-vinci-client/src/main/java/com/linkedin/davinci/DaVinciBackend.java

```diff
@@ -124,14 +124,16 @@ public DaVinciBackend(
 
       SchemaReader partitionStateSchemaReader = ClientFactory.getSchemaReader(
           ClientConfig.cloneConfig(clientConfig)
-              .setStoreName(AvroProtocolDefinition.PARTITION_STATE.getSystemStoreName()));
```

Comment:
I have a general question - I don't see in this diff how the passed in IC invoked? Could you explain how it is working under the new implementation?

Reply:
AFAIK IC provider is passed implicity when client is called from user applications. But when venice-server calls client, only then we need to set those up.